lib/pbench/server/api/resources/users_api.py

```diff
@@ -252,6 +233,25 @@ def post(self):
         except Exception:
             current_app.logger.exception("Exception while logging in a user")
             abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        else:
+            current_app.logger.debug(
+                "New auth token registered for user {}", user.username
+            )
+
+        # We take the opportunity to remove any expired tokens for ANY user
```

Comment:
Why do it during the login? I don't know the impact of it during the login process but I think the better place to remove all the expired tokens would be in the logout. 

I might also suggest if we are removing the expired tokens during the login we might do it only for the user who is trying to log in instead of the _Any_ user.

Reply:
We want this operation on a common but low frequency path.  Login is a nice reliable spot.  I don't see how logout would be better, and, for many users, they will probably just close the tab and never log out, so that wouldn't be a particularly reliable point in the flow.

As for removing expired tokens, an expired token is invalid no matter who it belongs to, and we can get them all with a single database operation, so why filter it to only the current user?

Reply:
The discussion is nearly moot as we expect our API keys to be permanent (at least until manually revoked through some API) and we don't hold or control Auth tokens from Keycloak. So this code has a short lifetime but might be valuable for our initial March deployment.

Reply:
People must login to do something, but they more often than not fail to logout.